### PR TITLE
Add documentation about GitHub Codespaces and GitHub.dev support

### PR DESCRIPTION
**Pull request by gabro**
This is in reference to https://github.com/scalameta/metals-feature-requests/issues/225

### CODE REVIEW

File: docs/editors/vscode.md [hunk 1]
```diff
@@ -271,3 +270,14 @@ extension to use default IntelliJ shortcuts with VS Code.
 | Type info        | Show hover                |
 | Expand           | Fold                      |
 | Extend Selection | Expand selection          |
+
+## GitHub Codespaces and GitHub.dev support
+
+First, some definitions:
+
+- [GitHub Codespaces] are remote dev containers you can connect to using VS Code Desktop or VS Code Web.
```
Comment (dos65) [inline docs/editors/vscode.md:278]:
```suggestion
- [GitHub Codespaces](https://github.com/features/codespaces) are remote dev containers you can connect to using VS Code Desktop or VS Code Web.
```

**Review by dos65 — APPROVED** [review]
Thx for clarification! I noticed the conversation in the original feature request and this thing is really confusing.

**Review by ckipp01 — COMMENTED** [review]
👍🏼 good to clarify this. One thing that comes to mind though is that we are talking about codespaces here, but then we also have an online IDEs section as well [here](https://scalameta.org/metals/docs/editors/online-ides#gitpod) where we talk about gitpod for example. Do you think we should try to centralize or link these?

File: docs/editors/vscode.md (hunk 1 above)
Comment (gabro) [reply docs/editors/vscode.md:278]: woops, thanks :) Integrating it

**Comment by gabro** [comment]
@ckipp01 that's a great idea! I've moved the documentation there and added a pointer from the vscode-specific one

File: docs/editors/vscode.md [hunk 2]
```diff
@@ -271,3 +270,7 @@ extension to use default IntelliJ shortcuts with VS Code.
 | Type info        | Show hover                |
 | Expand           | Fold                      |
 | Extend Selection | Expand selection          |
+
+## GitHub Codespaces and GitHub.dev support
+
+See https://scalameta.org/metals/docs/editors/online-ides#github-codespaces-and-githubdev
```
Comment (gabro) [inline docs/editors/vscode.md:276]: the domain is hardcoded, since we copy-paste this file to the VS Code README during the release process

**Review by ckipp01 — APPROVED** [review]
Looks good! Thanks @gabro!

**Comment by gabro** [comment]
Forcing the merge, since the CI failure is unrelated

**Pull request merged**